common/v2/services/ApiService/TokenInfo/TokenInfo.ts

```diff
@@ -21,14 +22,37 @@ export default class TokenInfoService {
   }
 
   public getTokensInfo = async (contractAddresses: string[]) => {
-    const params = new URLSearchParams();
-    contractAddresses.forEach(address => params.append('contractAddress', address));
+    // Splices contract addresses for large sets of contracts to break up the calls into manageable chunks
+    const splicedContractAddressArrays = spliceContractAddresses(contractAddresses);
+    const paramsArray = splicedContractAddressArrays.map(contractAddressArray =>
+      createParams(contractAddressArray)
+    );
 
     try {
-      const response = await this.service.get('', { params });
-      return response.data;
+      return await Promise.all(
+        paramsArray.map(async (params: any) => this.service.get('', { params }))
+      ).then((returnedArray: AxiosResponse[]) =>
+        _.union(...returnedArray.map(returnObject => returnObject.data))
+      );
     } catch (e) {
       throw e;
     }
   };
 }
+
+const spliceContractAddresses = (contractAddresses: string[]): string[][] => {
+  const lengthToSplitOn = 20;
+  const splicedContractAddressArrays = [];
+  do {
+    splicedContractAddressArrays.push([...contractAddresses.splice(0, lengthToSplitOn)]);
+  } while (contractAddresses.length >= lengthToSplitOn);
+
+  splicedContractAddressArrays.push([...contractAddresses]);
+  return splicedContractAddressArrays;
+};
```

Comment:
Naming:
This function is not specific to ContractAddresses and will work on any list.
To increase reuse you can add the length as an argument and use a more generic name such as `chunk`
https://medium.com/@Dragonza/four-ways-to-chunk-an-array-e19c889eac4

Ideally use [`R.splitEvery`](https://ramdajs.com/docs/#splitEvery)! or [`_.chunk`](https://lodash.com/docs/#chunk)